coordinator/src/main/java/com/tencent/rss/coordinator/AccessCandidatesChecker.java

```diff
@@ -0,0 +1,132 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AccessCandidatesChecker implements AccessChecker {
+  private static final Logger LOG = LoggerFactory.getLogger(AccessCandidatesChecker.class);
+
+  private final AtomicReference<Set<String>> candidates = new AtomicReference<>();
+  private final AtomicLong lastCandidatesUpdateMS = new AtomicLong(0L);
+  private final String path;
+  private final int updateIntervalS;
+
+  private ScheduledExecutorService updateAccessCandidatesSES;
+
+  static final String CRON_TASK_PARAM_DELIMITER = "_";
+
+  public AccessCandidatesChecker(AccessManager accessManager) {
+    CoordinatorConf conf = accessManager.getCoordinatorConf();
+    this.path = conf.get(CoordinatorConf.COORDINATOR_ACCESS_CANDIDATES_PATH);
+    this.updateIntervalS = conf.get(CoordinatorConf.COORDINATOR_ACCESS_CANDIDATES_UPDATE_INTERVAL_SEC);
+    init();
+  }
+
+  private void init() {
+    updateAccessCandidatesSES = Executors.newSingleThreadScheduledExecutor(
+        new ThreadFactoryBuilder().setDaemon(true).setNameFormat("UpdateAccessCandidates-%d").build());
+    updateAccessCandidatesSES.scheduleAtFixedRate(
+        this::updateAccessCandidates, 0, updateIntervalS, TimeUnit.SECONDS);
+  }
+
+  public AccessCheckResult check(String accessInfo) {
+    String taskId = checkAndExtractTaskId(accessInfo);
+    if (!candidates.get().contains(taskId)) {
+      String msg = String.format("Reject taskId[%s] access request.", taskId);
+      LOG.debug(msg);
+      return new AccessCheckResult(false, msg);
+    }
+
+    return new AccessCheckResult(true, "");
+  }
+
+  public void stop() {
+    if (updateAccessCandidatesSES != null) {
+      updateAccessCandidatesSES.shutdownNow();
+      updateAccessCandidatesSES = null;
+    }
+  }
+
+  private void updateAccessCandidates() {
+    try {
+      File candidatesFile = new File(path);
+      if (candidatesFile.exists()) {
+        long lastModifiedMS = candidatesFile.lastModified();
+        if (lastCandidatesUpdateMS.get() != lastModifiedMS) {
+          updateAccessCandidates(candidatesFile);
+          lastCandidatesUpdateMS.set(lastModifiedMS);
+        }
+      } else {
+        candidates.set(Sets.newConcurrentHashSet());
+      }
+      // TODO: add access num metrics
+    } catch (Exception e) {
+      LOG.warn("Error when update access candidates", e);
+    }
+  }
+
+  private void updateAccessCandidates(File candidatesFile) {
+    Set<String> newCandidates = Sets.newHashSet();
+    try (Stream<String> lines = Files.lines(candidatesFile.toPath())) {
+      lines.forEach(line -> {
+        String taskId = line.trim();
+        if (!StringUtils.isEmpty(taskId)) {
+          newCandidates.add(taskId);
+        }
+      });
+    } catch (Exception e) {
+      LOG.warn("Error when parse file {}", candidatesFile.getAbsolutePath(), e);
+    }
+    candidates.set(newCandidates);
+  }
+
+  @VisibleForTesting
+  String checkAndExtractTaskId(String cronTaskParam) {
```

Comment:
cronTaskParam ?

Reply:
Will fix it to align with last comment "use accessInfo"